_inc/lib/core-api/wpcom-endpoints/memberships.php

```diff
@@ -149,6 +205,50 @@ public function create_product( $request ) {
 		return $request;
 	}
 
+	/**
+	 * Automatically generate products according to type.
+	 *
+	 * @param object $request - request passed from WP.
+	 *
+	 * @return array|WP_Error
+	 */
+	public function create_products( $request ) {
+		if ( ( defined( 'IS_WPCOM' ) && IS_WPCOM ) ) {
+			jetpack_require_lib( 'memberships' );
+			$connected_destination_account_id = Jetpack_Memberships::get_connected_account_id();
+			if ( ! $connected_destination_account_id ) {
+				return new WP_Error( 'no-destination-account', __( 'Please set up a Stripe account for this site first', 'jetpack' ) );
+			}
+			return Memberships_Product::generate_default_products( get_current_blog_id(), $request['type'], $request['currency'], $connected_destination_account_id );
```

Comment:
This needs to be updated to return 400 codes rather than 500s in the case of invalid params.

Reply:
469875fb6bd039768e88328c885146ed7ea91fcd